edx_toggles/tests/test_waffle.py

```diff
@@ -0,0 +1,37 @@
+"""
+Unit tests for waffle classes.
+"""
+
+from django.test import TestCase
+
+# TODO import from edx_toggles.toggles once we remove the legacy classes from the exposed API
+from edx_toggles.toggles.internal.waffle.flag import WaffleFlag
+from edx_toggles.toggles.internal.waffle.switch import WaffleSwitch
+
+
+class WaffleFlagTests(TestCase):
+    """
+    WaffleFlag tests.
+    """
+
+    def test_name_validation(self):
+        WaffleFlag("namespaced.name", module_name="module1")
+        self.assertRaises(
+            ValueError, WaffleFlag, "non_namespaced", module_name="module1"
```

Comment:
Should we include the flag name, or do you think the stacktrace will be enough?

Reply:
The ValueError will include the non-namespaced flag name.